packages/@aws-cdk/aws-ec2/lib/auto-scaling-group.ts

```diff
@@ -191,4 +238,200 @@ export class AutoScalingGroup extends cdk.Construct implements IClassicLoadBalan
     public addToRolePolicy(statement: cdk.PolicyStatement) {
         this.role.addToPolicy(statement);
     }
+
+    /**
+     * Apply CloudFormation update policies for the AutoScalingGroup
+     */
+    private applyUpdatePolicies(props: AutoScalingGroupProps) {
+        if (props.updateType === UpdateType.ReplacingUpdate) {
+            this.asgUpdatePolicy.autoScalingReplacingUpdate = { willReplace: true };
+
+            if (props.replacingUpdateMinSuccessfulInstancesPercent !== undefined) {
+                if (this.autoScalingGroup.options.creationPolicy === undefined) {
+                    this.autoScalingGroup.options.creationPolicy = {};
+                }
+
+                // Yes, this goes on CreationPolicy, not as a process parameter to ReplacingUpdate.
+                // It's a little confusing, but the docs seem to explicitly state it will only be used
+                // during the update?
+                //
+                // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-creationpolicy.html
+                this.autoScalingGroup.options.creationPolicy.autoScalingCreationPolicy = {
```

Comment:
Right now I don't see support for `ResourceSignal`

```  
"ResourceSignal" : {    
    "Count" : Integer,
    "Timeout" : String
  }
```

We currently use this in deployments where if `userdata` does not complete the stack should fail. Specific example would be in our bastion pattern. Without this the stack can complete but not have completed the `userdata` aspects to enable ssh and other security checks.

Reply:
Ack, thanks!